modules/cudaoptflow/src/nvidiaOpticalFlow.cpp

```diff
@@ -0,0 +1,581 @@
+//
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+//M*/
+#include "precomp.hpp"
+#include "NVIDIAOpticalFlowSDK_1_0_Headers/NVIDIAOpticalFlowSDK-79c6cee80a2df9a196f20afd6b598a9810964c32/nvOpticalFlowCommon.h"
+#include "NVIDIAOpticalFlowSDK_1_0_Headers/NVIDIAOpticalFlowSDK-79c6cee80a2df9a196f20afd6b598a9810964c32/nvOpticalFlowCuda.h"
```

Comment:
With `ocv_include_directories()` it is enough to include:
```
#include "nvOpticalFlowCommon.h"
#include "nvOpticalFlowCuda.h"
```

Reply:
I thought the same and tried including like this
#include "nvOpticalFlowCommon.h"
#include "nvOpticalFlowCuda.h"
But compilation failed and hence i added them this way

Reply:
Include directory is not added properly.

Use this:
```
ocv_define_module(cudaoptflow opencv_video opencv_optflow opencv_cudaarithm opencv_cudawarping opencv_cudaimgproc OPTIONAL opencv_cudalegacy WRAP python)

set(NVIDIA_OPTICAL_FLOW_1_0_HEADERS_COMMIT "79c6cee80a2df9a196f20afd6b598a9810964c32")
set(NVIDIA_OPTICAL_FLOW_1_0_HEADERS_MD5 "ca5acedee6cb45d0ec610a6732de5c15")
set(NVIDIA_OPTICAL_FLOW_1_0_HEADERS_PATH "${OpenCV_BINARY_DIR}/3rdparty/NVIDIAOpticalFlowSDK_1_0_Headers")
ocv_download(FILENAME "${NVIDIA_OPTICAL_FLOW_1_0_HEADERS_COMMIT}.zip"
               HASH ${NVIDIA_OPTICAL_FLOW_1_0_HEADERS_MD5}
               URL
                 "https://github.com/NVIDIA/NVIDIAOpticalFlowSDK/archive/"
               DESTINATION_DIR "${NVIDIA_OPTICAL_FLOW_1_0_HEADERS_PATH}"
               STATUS NVIDIA_OPTICAL_FLOW_1_0_HEADERS_DOWNLOAD_SUCCESS
               ID "NVIDIA_OPTICAL_FLOW"
               UNPACK RELATIVE_URL)

if(NOT NVIDIA_OPTICAL_FLOW_1_0_HEADERS_DOWNLOAD_SUCCESS)
  message(STATUS "Failed to download NVIDIA_Optical_Flow_1_0 Headers")
else()
  add_definitions(-DHAVE_NVIDIA_OPTFLOW=1)
  ocv_include_directories(SYSTEM "${NVIDIA_OPTICAL_FLOW_1_0_HEADERS_PATH}/NVIDIAOpticalFlowSDK-${NVIDIA_OPTICAL_FLOW_1_0_HEADERS_COMMIT}")
endif()
```

Reply:
Also take a look on `add_definitions(-DHAVE_NVIDIA_OPTFLOW=1)` line.
You should properly guard implementation code (if downloading is failed/disabled/not available).

Reply:
Thanks alot :-)
I bounded implementation code as if headers are not downloaded then return HeaderIs Null error